touchforms/backend/xformserver.py

```diff
@@ -237,8 +237,15 @@ def handle_request(content, server):
             ensure_required_params(['session-id', 'ix'], action, content)
             return xformplayer.delete_repeat(content['session-id'], content['ix'], content.get('form_ix'))
         elif action == xformplayer.Actions.SUBMIT_ALL:
-            ensure_required_params(['session-id'], action, content)
-            return xformplayer.submit_form(content['session-id'], content.get('answers', []), content.get('prevalidated', False))
+            if content.get('session-id', None) and xformplayer.global_state:
+                xfsess = xformplayer.global_state.get_session(content['session-id'])
+                session_data = xfsess.orig_params['session_data']
+            else:
+                session_data = content.get('session-data')
+            return xformplayer.submit_form(content['session-id'],
```

Comment:
I actually got this from https://github.com/dimagi/touchforms/pull/142/files#diff-01433444d689961007b83189a7b1db1eR285 (moot now due to getting session_data from XFormSession tho)